from_dict/_from_dict.py

```diff
@@ -329,6 +342,8 @@ def _from_dict_inner(
         for arg, val in given_args.items():
             if arg not in ckwargs and arg not in created_object.__dict__:
                 created_object.__dict__[arg] = val
+    elif fd_error_on_unknown and given_args:
+        raise RuntimeError("arguments were supplied that are not required.")
```

Comment:
Let's also mention that fd_error_on_unknown is set to true and that's why we are complaining.

Can we also tell the user which arguments?

Nit: Could we make this a public special exception class such that callers can catch this exception?